src/create.rs

```diff
@@ -409,13 +428,58 @@ pub struct Execution {
     locale: Option<PathBuf>,
     name: Option<String>,
     no_build: bool,
+    wix_toolset: WixToolset,
+    wix_toolset_upgrade: WixToolsetUpgrade,
     install: bool,
     output: Option<String>,
     package: Option<String>,
     target: Option<String>,
     version: Option<String>,
 }
 
+/// Enumeration of wix-toolset options
+/// 
+/// This controls which wix build binaries are used by cargo-wix
+#[derive(ValueEnum, Copy, Clone, Debug)]
+pub enum WixToolset {
+    /// The default wix toolset uses "candle.exe" and "light.exe" to build the installer
+    Default,
+    /// Modern wix toolsets use just "wix.exe" to build the installer
+    Modern,
+}
+
+/// Enumeration of wix-toolset upgrade patterns that can be applied
+/// 
+/// WiX toolset upgrading consists of two parts,
+/// 
+/// 1) Convert Wix3 and below *.wxs files to the modern format
+/// 2) Detect and install wix extensions required by *.wxs files
+#[derive(ValueEnum, Copy, Clone, Debug)]
+pub enum WixToolsetUpgrade {
+    /// Do not apply any wix toolset upgrade logic
+    None,
+    /// Upgrade files in place, install extensions globally
+    #[clap(name = "inplace")]
+    Inplace,
+    /// Upgrade files in place, but use "vendoring" when installing wix extensions.
+    /// 
+    /// This will install wix extensions in the current directory.
+    #[clap(name = "inplace-vendor")]
+    InplaceVendored,
+    /// Upgrade source files in a SxS manner
+    /// 
+    /// This will copy *.wxs files to a versioned folder and continue to use that folder to install wix extensions.
+    #[clap(name = "sxs")]
+    SideBySide,
+}
+
+impl WixToolset {
+    /// Returns true if the toolset in use is modern
+    pub fn is_modern(&self) -> bool {
```

Comment:
Further down, the `!toolset.is_modern` is used frequently. Please also add a `is_legacy` (or whatever the name of the variant ends up being, see previous comment) method and let's use that instead of negating the modern variant on each usage. It will be easier to follow and read in the future.